app/src/main/java/com/door43/translationstudio/core/TargetTranslation.java

```diff
@@ -376,24 +382,51 @@ public void addSourceTranslation(SourceTranslation sourceTranslation) throws JSO
         boolean foundDuplicate = false;
         for(int i = 0; i < sourceTranslationsJson.length(); i ++) {
             JSONObject obj = sourceTranslationsJson.getJSONObject(i);
-            if(obj.getString("language_id").equals(sourceTranslation.sourceLanguageSlug) && obj.getString("resource_id").equals(sourceTranslation.resourceSlug)) {
+            if(obj.getString(LANGUAGE_ID).equals(sourceTranslation.sourceLanguageSlug) && obj.getString(RESOURCE_ID).equals(sourceTranslation.resourceSlug)) {
                 foundDuplicate = true;
                 break;
             }
         }
         if(!foundDuplicate) {
             JSONObject translationJson = new JSONObject();
-            translationJson.put("language_id", sourceTranslation.sourceLanguageSlug);
-            translationJson.put("resource_id", sourceTranslation.resourceSlug);
-            translationJson.put("checking_level", sourceTranslation.getCheckingLevel());
-            translationJson.put("date_modified", sourceTranslation.getDateModified());
-            translationJson.put("version", sourceTranslation.getVersion());
+            translationJson.put(LANGUAGE_ID, sourceTranslation.sourceLanguageSlug);
+            translationJson.put(RESOURCE_ID, sourceTranslation.resourceSlug);
+            translationJson.put(CHECKING_LEVEL, sourceTranslation.getCheckingLevel());
+            translationJson.put(DATE_MODIFIED, sourceTranslation.getDateModified());
+            translationJson.put(VERSION, sourceTranslation.getVersion());
             sourceTranslationsJson.put(translationJson);
             manifest.put(FIELD_SOURCE_TRANSLATIONS, sourceTranslationsJson);
         }
     }
 
     /**
+     * get sources list from manifest
+     */
+    public String[] getSourceTranslations() {
+
+        try {
+            List<String> sources = new ArrayList<>();
+
+            JSONArray sourceTranslationsJson = manifest.getJSONArray(FIELD_SOURCE_TRANSLATIONS);
+
+            for (int i = 0; i < sourceTranslationsJson.length(); i++) {
+                JSONObject obj = sourceTranslationsJson.getJSONObject(i);
+
+                String sourceLanguageSlug = obj.getString(LANGUAGE_ID);
+                String resourceSlug = obj.getString(RESOURCE_ID);
+
+                SourceTranslation sourceTranslation =  SourceTranslation.simple(this.projectId,sourceLanguageSlug,resourceSlug);
+                sources.add(sourceTranslation.getId());
+            }
+
+            return sources.toArray(new String[sources.size()]);
+        } catch(Exception e) {
+            Logger.e(TAG, "Error reading sources", e);
+        }
+        return null;
```

Comment:
this method returns a list so rather than returning null we should return an empty list.